sdk/core/src/metadata.rs

```diff
@@ -0,0 +1,34 @@
+use crate::AddAsHeader;
+use http::request::Builder;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+pub struct Metadata(HashMap<String, String>);
+
+impl Metadata {
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    pub fn as_mut(&mut self) -> &mut HashMap<String, String> {
```

Comment:
Maybe we can provide an `insert` for convenience since I imagine 99% of the time people will call `as_mut().insert(key, value)`